ffi/dotnet/Devolutions.IronRdp/src/Framed.cs

```diff
@@ -16,12 +17,16 @@ public Framed(S stream)
         return (this.stream, this.buffer);
     }
 
-    public byte[] Peek()
+    public Span<byte> Peek()
     {
-        return this.buffer.ToArray();
+        return CollectionsMarshal.AsSpan(this.buffer);
     }
```

Comment:
suggestion: Document that items should not be added or removed from the List while the Span is in use. This means no reads and no writes.

Reply:
updated!